client/service/test/src/client/mod.rs

```diff
@@ -1443,32 +1441,51 @@ fn respects_block_rules() {
 				.build()
 		};
 
+		// test modus operandi:
+		//
+		//            B[2]
+		//           /
+		// G[0]--B[1]
+		//     \     \
+		//      \     B'[2]
+		//       \
+		//        B'[1]
+		//
+		// B[1]  - block ok
+		// B'[1] - block not ok, added to block_rules::bad
+		//
+		// B[2]  - block ok, correct fork for block height==2, added to block_rules::forks
+		// B'[2] - block not ok, (incorrect fork)
+
+		// build B[1]
 		let block_ok = client
-			.new_block_at(&BlockId::Number(0), Default::default(), false)
+			.new_block_at(client.chain_info().genesis_hash, Default::default(), false)
 			.unwrap()
 			.build()
 			.unwrap()
 			.block;
+		let block_ok_1_hash = block_ok.hash();
 
 		let params = BlockCheckParams {
 			hash: block_ok.hash(),
-			number: 0,
+			number: 1,
```

Comment:
Actually number given here does not matter, but the code is less confusing if the number actually matches the block.